### PR TITLE
cleaning up logs a bit

### PR DESCRIPTION
**Pull request by shivasishdas**
Will do it at other places as well once the refactor PRs are merged.

### CODE REVIEW

File: internal/subscriber/subscriber.go [hunk 1]
```diff
@@ -541,9 +554,18 @@ func (s *Subscriber) logInMemoryStats() {
 		}
 		st[tp.String()] = total
 	}
+	logger.Ctx(s.ctx).Infow("subscriber: in-memory stats", s.getLogFields(), "stats", st)
+}
 
-	logger.Ctx(s.ctx).Infow("subscriber: in-memory stats", "stats", st,
-		"topic", s.topic, "subscription", s.subscription, "subscriberId", s.subscriberID)
+// returns a map of common fields to be logged
+func (s *Subscriber) getLogFields() map[string]interface{} {
```
Comment (sunnyagg) [inline internal/subscriber/subscriber.go:561]: can we set them in context ?

Comment (shivasishdas) [reply internal/subscriber/subscriber.go:561]: Had tried this. Values set in `ctx()` were not getting goroutine scoped. eg: `msgID` added to `ctx` of one request as getting used up in consecutive requests as well. Maybe I was doing it wrong. Looking for implementation suggestions using this.

File: internal/credentials/model.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ func (m *Model) Prefix() string {
 }
 
 // NewCredential returns a new credential model
-func NewCredential(username, password string) ICredentials {
+func NewCredential(username, password string) *Model {
```
Comment (sunnyagg) [inline internal/credentials/model.go:62]: IModel ?

Comment (shivasishdas) [reply internal/credentials/model.go:62]: Serde was failing with the interface previously hence changed it to the struct{}.

**Review by sunnyagg — APPROVED** [review]

**Pull request merged**